tests/Kernel/MarkupRenderingTest.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\Kernel;
+
+use Drupal\Core\Form\FormInterface;
+use Drupal\Core\Form\FormState;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+use Drupal\Core\Serialization\Yaml;
+use Drupal\Core\Site\Settings;
+use Drupal\KernelTests\KernelTestBase;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests components and patterns markup.
+ *
+ * This kernel test is heavily inspired by OpenEuropa Theme (oe_theme) similar
+ * test: \Drupal\Tests\oe_theme\Kernel\RenderingTest.
+ *
+ * @see https://github.com/openeuropa/oe_theme/blob/HEAD/tests/Kernel/RenderingTest.php
+ *
+ * @group oe_bootstrap_theme
+ */
+class MarkupRenderingTest extends KernelTestBase implements FormInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'components',
+    'filter',
+    'oe_bootstrap_theme_helper',
+    'system',
+    'ui_patterns',
+    'ui_patterns_library',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Replicate 'file_scan_ignore_directories' from settings.php.
+    $settings = Settings::getAll();
+    $settings['file_scan_ignore_directories'] = [
+      'node_modules',
+      'bower_components',
+      'vendor',
+      'build',
+      'bcl',
+    ];
+    new Settings($settings);
+
+    $this->container->get('theme_installer')->install(['oe_bootstrap_theme']);
+    $this->config('system.theme')->set('default', 'oe_bootstrap_theme')->save();
+    $this->container->set('theme.registry', NULL);
+  }
+
+  /**
+   * Tests rendering of elements.
+   *
+   * @param array $render_array
+   *   A render array.
+   * @param array $expectations
+   *   Test assertion expectations.
+   *
+   * @dataProvider markupRenderingProvider
+   */
+  public function testMarkupRendering(array $render_array, array $expectations): void {
+    // Wrap all the test structure inside a form. This will allow proper
+    // processing of form elements and invocation of form alter hooks. Even if
+    // the elements being tested are not form related, the form can host them
+    // without causing any issues.
+    $form_state = new FormState();
+    $form_state->addBuildInfo('args', [$render_array]);
+    $form_state->setProgrammed();
+
+    $form = $this->container->get('form_builder')->buildForm($this, $form_state);
+    $this->assertMarkupRendering($expectations, (string) $this->container->get('renderer')->renderRoot($form));
+  }
+
+  /**
+   * Data provider for markup rendering tests.
+   *
+   * @return array
+   *   A set of dump data for testing.
+   *
+   * @see self::testMarkupRendering()
+   * @see tests/fixtures/markup_rendering.yml
+   */
+  public function markupRenderingProvider(): array {
+    return Yaml::decode(file_get_contents(__DIR__ . '/../fixtures/markup_rendering.yml'));
```

Comment:
As we've been discussed in the daily, let's create 2 YAML files: `markup_rendering.yml` & `markup_rendering_pattern.yml`. So, that we'll create all `'#type': pattern` in the later. Here, we need to merge them in order to return all.